src/main/kotlin/no/nav/helse/spenn/rest/api/v1/OppdragStateController.kt

```diff
@@ -13,42 +15,43 @@ import java.util.*
 // NB: Sync tilgangsstyring med helse-spade
 @Protected
 @RestController
-@RequestMapping("/api/v1")
+@RequestMapping("/api/v1/oppdrag")
 class OppdragStateController(val oppdragStateService: OppdragStateService,
-                             val oidcRequestContextHolder: OIDCRequestContextHolder,
-                             @Value("\${api.access.requiredgroup:group1}") val requiredGroupMembership: String ) {
+                             val oidcRequestContextHolder: OIDCRequestContextHolder) {
 
     companion object {
         private val LOG = LoggerFactory.getLogger(OppdragStateController::class.java)
         private val AUDIT_LOG = LoggerFactory.getLogger("auditLogger")
     }
 
-    @GetMapping("/oppdrag/soknad/{soknadId}")
+    @GetMapping("/soknad/{soknadId}")
     fun getOppdragStateBySoknadId(@PathVariable soknadId: UUID): OppdragStateDTO {
-        validateGroupMembership()
         LOG.info("Rest retrieve for soknadId: ${soknadId}")
         AUDIT_LOG.info("Bruker=${currentNavIdent()} slår opp søknadId=${soknadId}")
         return oppdragStateService.fetchOppdragState(soknadId)
     }
 
-    @GetMapping("/oppdrag/{id}")
+    @GetMapping("/{id}")
     fun getOpppdragStateById(@PathVariable id: Long): OppdragStateDTO {
-        validateGroupMembership()
         LOG.info("Rest retrieve for id: ${id}")
         AUDIT_LOG.info("Bruker=${currentNavIdent()} slår opp oppdragId=${id}")
         return oppdragStateService.fetchOppdragStateById(id)
     }
 
-    private fun validateGroupMembership() {
-        if (currentUserGroups() == null || (!currentUserGroups().contains(requiredGroupMembership))) {
-            LOG.debug("${currentNavIdent()} prøvde accessere API men mangler gruppen $requiredGroupMembership")
-            throw MissingGroupException("Missing group $requiredGroupMembership in JWT")
-        }
+    @PutMapping("/{id}")
+    fun updateOppdragState(@PathVariable id: Long, @RequestBody dto: OppdragStateDTO): OppdragStateDTO {
+        LOG.info("Rest update for id: ${id}")
+        AUDIT_LOG.info("Bruker=${currentNavIdent()} oppdaterer oppdragId=${id}")
+        return oppdragStateService.saveOppdragState(dto)
```

Comment:
Er vi litt vel fleksible her med hva man kan tweake på?  Og kanskje vi i såfall burde logge historikk?

Reply:
auditloggen burde holder i første omgang, jeg tenker denne er ment for krisehåndtering av oss utviklerer :)